src/components/ArticlePage.jsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { format } from "date-fns";
+import { fetchSingleArticle } from "../../utils";
+import "../css/ArticlePage.css";
+
+const ArticlePage = () => {
+  const { article_id } = useParams();
+  const [currArticle, setCurrArticle] = useState({});
+
+  useEffect(() => {
+    fetchSingleArticle(article_id)
+      .then((article) => {
+        setCurrArticle(article);
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
We could also have an Error state that is false and becomes true when an error is caught, this would allow you to render something specific in those instances 🙂